### ISSUE TITLE
CocoaPods 1.7.0beta3 regression with lottie-ios modulemap when using static linking

### ISSUE THREAD

**Issue by chrisballinger** [CocoaPods/CocoaPods#8695]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Update our Gemfile to point from CocoaPods 1.6.1 to 1.7.0beta3.

My guess is that this regression is related to lottie's use of custom module names conflicting with how CocoaPods handles Swift static library module maps.

Btw the issue also affects both Lottie 2.5.3 (Obj-C) and Lottie 3 rewrite.

## What did you expect to happen?

All dependencies previously compile-able are still compiled.

## What happened instead?

lottie-ios can no longer be compiled.

<unknown>:0: error: module map file `.../app/Pods/Headers/Public/Lottie/lottie-ios.modulemap' not found
<unknown>:0: error: underlying Objective-C module 'Lottie' not found

### Stack

```
   CocoaPods : 1.7.0.beta.3
        Ruby : ruby 2.6.1p33 (2019-01-30 revision 66950) [x86_64-darwin18]
    RubyGems : 3.0.2
        Host : Mac OS X 10.14.4 (18E226)
       Xcode : 10.2 (10E125)
         Git : git version 2.21.0
Ruby lib dir : /usr/local/Cellar/ruby/2.6.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 4db5f7fd60ba774deb000ec442f861594a50b460
```

### Installation Source

```
Executable Path: /usr/local/lib/ruby/gems/2.6.0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
platform :ios, '11.0'
inhibit_all_warnings!

target 'Regression' do
  pod 'lottie-ios'
end
```

## Project that demonstrates the issue

~I will try to make an isolated project test case soon.~ There is an isolated test case project demonstrating the issue here: https://github.com/chrisballinger/cocoapods-issue-8695

**Comment by segiddins** [comment]
Does this happen even without the post-install hook?

**Comment by chrisballinger** [comment]
ya it's still broken, with or without it. my main reason for trying 1.7 was that it contains the fix for #8174

**Comment by chrisballinger** [comment]
@segiddins I made an example repo isolating the regression here: https://github.com/chrisballinger/cocoapods-issue-8695

The 1.7.0 folder exhibits the regression, the 1.6.1 folder is an example of it working.

My initial guess was that it had something to do with lottie-ios's customization of the module name, but I tried commenting out those lines of the podspec without success.

**Comment by chrisballinger** [comment]
I have also confirmed this only affects integration as a static library, not as a framework.

**Comment by chrisballinger** [comment]
The issue seems to be the dash in the pod name. Changing the podspec name to `Lottie` from `lottie-ios` to the following resolves the issue:

# Works
```ruby
Pod::Spec.new do |s|
  s.name             = 'Lottie'
  s.version          = '3.0.4'
  s.summary          = 'A library to render native animations from bodymovin json. Now in Swift!'

s.description      = <<-DESC
Lottie is a mobile library for Android and iOS that parses Adobe After Effects animations exported as json with bodymovin and renders the vector animations natively on mobile and through React Native!

For the first time, designers can create and ship beautiful animations without an engineer painstakingly recreating it be hand. Since the animation is backed by JSON they are extremely small in size but can be large in complexity! Animations can be played, resized, looped, sped up, slowed down, and even interactively scrubbed.
DESC

  s.homepage         = 'https://github.com/airbnb/lottie-ios'
  s.license          = { :type => 'Apache', :file => 'LICENSE' }
  s.author           = { 'Brandon Withrow' => 'buba447@gmail.com' }
  s.source           = { :git => 'https://github.com/airbnb/lottie-ios.git', :tag => s.version.to_s }
  
  s.swift_version = '4.2'
  s.ios.deployment_target = '9.0'
  s.osx.deployment_target = '10.10'
  s.tvos.deployment_target = '9.0'
  
  s.source_files = 'lottie-swift/src/**/*'
  s.ios.source_files   = 'lottie-swift/iOS/*.swift'
  s.ios.exclude_files   = 'lottie-swift/src/Public/MacOS/**/*'
  s.tvos.exclude_files   = 'lottie-swift/src/Public/MacOS/**/*'
  s.osx.exclude_files = 'lottie-swift/src/Public/iOS/**/*'
  
  s.ios.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.tvos.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.osx.frameworks = ['AppKit', 'CoreGraphics', 'QuartzCore']
  s.header_dir = 'Lottie'
end
```

# Doesn't work

```
Pod::Spec.new do |s|
  s.name             = 'lottie-ios'
  s.version          = '3.0.4'
  s.summary          = 'A library to render native animations from bodymovin json. Now in Swift!'

s.description      = <<-DESC
Lottie is a mobile library for Android and iOS that parses Adobe After Effects animations exported as json with bodymovin and renders the vector animations natively on mobile and through React Native!
For the first time, designers can create and ship beautiful animations without an engineer painstakingly recreating it be hand. Since the animation is backed by JSON they are extremely small in size but can be large in complexity! Animations can be played, resized, looped, sped up, slowed down, and even interactively scrubbed.
DESC

  s.homepage         = 'https://github.com/airbnb/lottie-ios'
  s.license          = { :type => 'Apache', :file => 'LICENSE' }
  s.author           = { 'Brandon Withrow' => 'buba447@gmail.com' }
  s.source           = { :git => 'https://github.com/airbnb/lottie-ios.git', :tag => s.version.to_s }
  
  s.swift_version = '4.2'
  s.ios.deployment_target = '9.0'
  s.osx.deployment_target = '10.10'
  s.tvos.deployment_target = '9.0'
  
  s.source_files = 'lottie-swift/src/**/*'
  s.ios.source_files   = 'lottie-swift/iOS/*.swift'
  s.ios.exclude_files   = 'lottie-swift/src/Public/MacOS/**/*'
  s.tvos.exclude_files   = 'lottie-swift/src/Public/MacOS/**/*'
  s.osx.exclude_files = 'lottie-swift/src/Public/iOS/**/*'
  
  s.ios.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.tvos.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.osx.frameworks = ['AppKit', 'CoreGraphics', 'QuartzCore']
  s.module_name = 'Lottie'
  s.header_dir = 'Lottie'
end
```

**Comment by amorde** [comment]
Dug into this a bit, the symlinked module map is getting deleted at some point. Still trying to find the cause of that

**Comment by amorde** [comment]
found it - `SandboxDirCleaner` is deleting it because that directory is normally used for public headers, and since this pod has none it was assumed this can be deleted. Will have to rework this a bit

**Comment by chrisballinger** [comment]
@amorde Wow, thanks for getting to the bottom of it!

**Comment by amorde** [comment]
Very detailed bug report + great example project helped a ton here :)

**Comment by dnkoutso** [comment]
@amorde this was fixed by #8706 right?

**Comment by amorde** [comment]
Yup! Guess Github didn’t catch it when the PR merged

**Comment by pluckq** [comment]
Reproduces with AsyncDisplayKit/Texture framework

**Comment by Coeur** [comment]
@pluckq can you try with the master branch of CocoaPods? A fix related to that issue was merged a few hours earlier.

**Comment by pluckq** [comment]
@Coeur, thank you for your replay, but I already were on stable 1.7.0

**Comment by Coeur** [comment]
@pluckq yes, when I wrote "a few hours earlier", it's about a fix that was merged AFTER the 1.7.0 release: https://github.com/CocoaPods/CocoaPods/pull/8837. I do not know if it solves your problem. If not, then please open a new issue.
